src/Method/Payment/Create.php

```diff
@@ -146,4 +93,72 @@ private function getPaymentInstrumentConstraints($method)
             )
         );
     }
+
+    private function getTrheeDS2DataConstraints()
+    {
+        return new Assert\Collection([
+            'notificationUrl' => $this->buildElement('string', 1),
+            'browserInfo' => $this->getBrowserInfoConstraints(),
+            'billingAddress' => $this->getAdressConstraints('billingAddress'),
+            'deliveryAddress' => $this->getAdressConstraints('deliveryAddress'),
+            'cardHolderInfo' => $this->getCardHolderInfoConstraints(),
+        ]);
+    }
+
+    private function getBrowserInfoConstraints()
+    {
+        return new Assert\Collection([
+            'acceptHeader' => $this->buildElement('string', 1),
+            'browserLanguage' => $this->buildElement('string', 1), // TODO should implement standard IETF BCP 47 (https://tools.ietf.org/html/bcp47)
+            'screenWidth' => $this->buildElement('integer', 1),
+            'screenHeight' => $this->buildElement('integer', 1),
+            'challengeWindowSize' => $this->buildElement('string', 1),
+            'userAgent' => $this->buildElement('string', 1),
+            'colorDepth' => $this->buildElement('integer', 1),
+            'timeZone' => $this->buildElement('integer', 1),
+            'ipAddress' => $this->buildElement('string'),
+            'javaEnabled' => $this->buildElement('bool'),
+            'javascriptEnabled' => $this->buildElement('bool'),
+        ]);
+    }
+
+    private function getAdressConstraints()
+    {
+        return new Assert\Collection([
+            'address_line1' => $this->buildElement('string', 1, ['max'=>50]),
+            'address_line2' => $this->buildElement('string', 1, ['max'=>50]),
+            'address_line3' => $this->buildElement('string', 0, ['max'=>50]),
+            'city' => $this->buildElement('string', 1, ['max'=>50]),
+            'country' => $this->buildElement('string', 1, ['max'=>10]), // TODO ISO 3166-1 alpha-2 country code.
+            'postal_code' => $this->buildElement('string', 1, ['max'=>16]),
+            'state' => $this->buildElement('string', 0, ['max'=>14]), // TODO  country subdivision code defined in ISO 3166-2
+        ]);
+    }
+
+    private function getCardHolderInfoConstraints()
+    {
+        return new Assert\Collection([
+            'email_address' => new Assert\Optional([
+                new Assert\Email(['mode'=>'loose'])
+            ]),
+            'mobile_phone_number' => $this->buildElement('string'),
+            'work_phone_number' => $this->buildElement('string'),
+            'home_phone_number' => $this->buildElement('string'),
+        ]); 
+    }
+
+    private function buildElement($typeValue, $isRequired=0, $length=0, $args=0)
```

Comment:
Nice refactoring of assertions :) I would consider using boolean instead of integers for $isRequired variable - perhaps that would be more descriptive